app/controllers/search_controller.rb

```diff
@@ -1,8 +1,22 @@
 class SearchController < ApplicationController
+  include SearchHelper
   def search
     session[:address] = params[:search] if params.has_key?(:search)
-    @search = Listing.where(is_active: true)
-    @search = @search.near("#{session[:address]}", 5, order: 'distance')
+    if check_if_filled?(session[:address])
+      @search = Listing.where(is_active: true)
+    else
+      @bookings = Booking.where("check_in_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'
+                                AND check_out_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'")
+                          .pluck(:listing_id)
+      @search = Listing.where(is_active: true).where.not(id: @bookings).near("#{session[:address]}", 5, order: 'distance')
+    end
+
+    if check_if_filled?(params[:check_in_date]) && check_if_filled?(params[:check_out_date])
+      @bookings = Booking.where("check_in_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'
+                                AND check_out_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'")
+                        .pluck(:listing_id)
+      @search = Listing.where(is_active: true).where.not(id: @bookings)
```

Comment:
Please make simplify this part.
Avoid repeating the same line of code in one method.

CLUE: To make code shorter, you can base it on the `check_conflicts` method for booking function.